test/views/components/RadiosWithSubHeadingSpec.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.components
+
+import org.jsoup.nodes.Element
+
+import play.api.i18n.{Lang, Messages, MessagesApi}
+import play.api.inject.Injector
+import play.api.test.FakeRequest
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.data.FormError
+import views.html.components.radiosWithSubHeading
+import play.twirl.api.Html
+import org.jsoup.nodes.{Document, Element}
+import scala.collection.JavaConverters._
+import views.ViewSpecBase
+
+class RadiosWithSubHeadingSpec extends ViewSpecBase with GuiceOneAppPerSuite {
+
+  val page = "/url"
+  val processId = "ext90087"
+  val fieldName = "field"
+  val heading = "Some Heading"
+  val linkText = "Link Text"
+
+  trait Test {
+    private def injector: Injector = app.injector
+    def messagesApi: MessagesApi = injector.instanceOf[MessagesApi]
+    implicit def messages: Messages = messagesApi.preferred(Seq(Lang("en")))
+  }
+
+  trait WelshTest extends Test {
+    implicit override def messages: Messages = messagesApi.preferred(Seq(Lang("cy")))
+  }
+
+  "radiosWithSubHeading generated Html" must {
+
+    def radios: radiosWithSubHeading = injector.instanceOf[radiosWithSubHeading]
+    val html: Html = radios(None,
+                            Seq.empty,
+                            page,
+                            processId,
+                            fieldName,
+                            heading,
+                            linkText)
+    val doc = asDocument(html)
```

Comment:
There are a few unused vals here. If they are not being used I would remove them.

Reply:
will remove